internal/core/metadatahandle.cpp

```diff
@@ -82,11 +82,12 @@ Axis make_single_cube_axis(
     );
 }
 
-SingleMetadataHandle::SingleMetadataHandle(OpenVDS::VolumeDataLayout const* const layout)
-    : m_layout(layout),
-      m_iline(make_single_cube_axis(layout, get_dimension({std::string(OpenVDS::KnownAxisNames::Inline())}))),
-      m_xline(make_single_cube_axis(layout, get_dimension({std::string(OpenVDS::KnownAxisNames::Crossline())}))),
-      m_sample(make_single_cube_axis(layout, get_dimension({std::string(OpenVDS::KnownAxisNames::Sample()), std::string(OpenVDS::KnownAxisNames::Depth()), std::string(OpenVDS::KnownAxisNames::Time())}))),
+SingleMetadataHandle::SingleMetadataHandle(
+    OpenVDS::VolumeDataLayout const* const layout,
+    std::unordered_map<AxisType, Axis> axes_map
```

Comment:
I completely agree that map looks better. My concern is rather that we add heap allocations, which we know is a potential bottleneck since there are a lot of heap allocations already happening that are outside our control. 

My suggestion is probably a premature optimisation though, and it makes more sense to introduce it in a separate PR and do some measurements to see if it makes a significant difference.

Reply:
There is 1 axes_map with 3 axes per user request that are done before any heavy work starts. I don't see where the bottleneck might come from.

Reply:
Agree, should be okay